src/models/exhaust-export-csv.ts

```diff
@@ -0,0 +1,121 @@
+import * as fs from 'fs/promises';
+import {ERRORS} from '../util/errors';
+import {ExhaustPluginInterface} from './exhaust-plugin-interface';
+const {InputValidationError, WriteFileError} = ERRORS;
+
+export const ExhaustExportCsv = (config: any): ExhaustPluginInterface => {
+  const extractConfigParams = () => {
+    const outputPath: string =
+      'output-path' in config
+        ? config['output-path']
+        : (() => {
+            throw new InputValidationError("Config does not have 'outputPath'");
+          })();
+    return [outputPath];
+  };
+
+  const handleLeafValue = (
```

Comment:
please provide JSDoc style documentation for each method to make clear what function does without reading the code.